src/tbb/def/win32-mingw-tbb.def

```diff
@@ -0,0 +1,143 @@
+; Copyright (c) 2005-2020 Intel Corporation
```

Comment:
@AlexVeprev, @longnguyen2004 , the def files look very similar with Linux *def files. Can we unify them somehow?

Reply:
The `win32-mingw` file is similar to `lin32`, with `char` replaced with `wchar_t`. Likewise, `win64-mingw` is similar to `lin64`, with `char` replaced with `wchar_t` and `(unsigned) long` replaced with `(unsigned) long long`

Reply:
By the way, how are you generating the MSVC and Linux def files? We can mix a bit between the two.

Reply:
> By the way, how are you generating the MSVC and Linux def files? We can mix a bit between the two.

Manually, that is the main burden issue.

Reply:
> The `win32-mingw` file is similar to `lin32`, with `char` replaced with `wchar_t`. Likewise, `win64-mingw` is similar to `lin64`, with `char` replaced with `wchar_t` and `(unsigned) long` replaced with `(unsigned) long long`

Interestingly, can we build mingw files from linux files on fly?

Reply:
It's certainly possible, but can it be done with pure CMake?
Here's what needs to be changed from the Linux def:
`lin32-tbb.def` (`char --> wchar_t`):
```
_ZN3tbb6detail2r115create_itt_syncEPvPKcS4_ --> _ZN3tbb6detail2r115create_itt_syncEPvPKwS4_
_ZN3tbb6detail2r117itt_set_sync_nameEPvPKc  --> _ZN3tbb6detail2r117itt_set_sync_nameEPvPKw
```

`lin64-tbb.def` (`char --> wchar_t`, `(unsigned) long --> (unsigned) long long`):
```
_ZN3tbb6detail2r115create_itt_syncEPvPKcS4_ --> _ZN3tbb6detail2r115create_itt_syncEPvPKwS4_
_ZN3tbb6detail2r117itt_set_sync_nameEPvPKc --> _ZN3tbb6detail2r117itt_set_sync_nameEPvPKw
_ZN3tbb6detail2r115allocate_memoryEm --> _ZN3tbb6detail2r115allocate_memoryEy
_ZN3tbb6detail2r122cache_aligned_allocateEm --> _ZN3tbb6detail2r122cache_aligned_allocateEy
_ZN3tbb6detail2r18allocateERPNS0_2d117small_object_poolEm --> _ZN3tbb6detail2r18allocateERPNS0_2d117small_object_poolEy
_ZN3tbb6detail2r18allocateERPNS0_2d117small_object_poolEmRKNS2_14execution_dataE --> _ZN3tbb6detail2r18allocateERPNS0_2d117small_object_poolEyRKNS2_14execution_dataE
_ZN3tbb6detail2r110deallocateERNS0_2d117small_object_poolEPvm --> _ZN3tbb6detail2r110deallocateERNS0_2d117small_object_poolEPvy
_ZN3tbb6detail2r110deallocateERNS0_2d117small_object_poolEPvmRKNS2_14execution_dataE --> _ZN3tbb6detail2r110deallocateERNS0_2d117small_object_poolEPvyRKNS2_14execution_dataE
_ZN3tbb6detail2r114notify_waitersEm --> _ZN3tbb6detail2r114notify_waitersEy
_ZN3tbb6detail2r16submitERNS0_2d14taskERNS2_18task_group_contextEPNS1_5arenaEm --> _ZN3tbb6detail2r16submitERNS0_2d14taskERNS2_18task_group_contextEPNS1_5arenaEy
_ZN3tbb6detail2r120isolate_within_arenaERNS0_2d113delegate_baseEl --> _ZN3tbb6detail2r120isolate_within_arenaERNS0_2d113delegate_baseEx
_ZN3tbb6detail2r18finalizeERNS0_2d121task_scheduler_handleEl --> _ZN3tbb6detail2r18finalizeERNS0_2d121task_scheduler_handleEx
_ZN3tbb6detail2r117parallel_pipelineERNS0_2d118task_group_contextEmRKNS2_11filter_nodeE --> _ZN3tbb6detail2r117parallel_pipelineERNS0_2d118task_group_contextEyRKNS2_11filter_nodeE
_ZN3tbb6detail2r126allocate_bounded_queue_repEm --> _ZN3tbb6detail2r126allocate_bounded_queue_repEy
_ZN3tbb6detail2r126wait_bounded_queue_monitorEPNS1_18concurrent_monitorEmlRNS0_2d113delegate_baseE --> _ZN3tbb6detail2r126wait_bounded_queue_monitorEPNS1_18concurrent_monitorEyxRNS0_2d113delegate_baseE
_ZN3tbb6detail2r128deallocate_bounded_queue_repEPhm --> _ZN3tbb6detail2r128deallocate_bounded_queue_repEPhy
_ZN3tbb6detail2r128notify_bounded_queue_monitorEPNS1_18concurrent_monitorEmm --> _ZN3tbb6detail2r128notify_bounded_queue_monitorEPNS1_18concurrent_monitorEyy
```
There's also `tbbmalloc`, and I'm working on it.
Basically everything using `tchar`, `std::size_t`, `std::(u)intptr_t` and `std::ptrdiff_t` have to be replaced for 64 bit.

Reply:
Also this is the [Python script](https://pastebin.com/tbrFGUWB) I use to create the def file from the Linux def file. I haven't implemented the symbol replacement yet, but it can be done with minimal effort.

Reply:
If we have specific rules for replacements, I'd agree on using script on fly, but I'd prefer to make implementation in CMake to avoid additional dependency on Python. It can be done on our side later.